### PR TITLE
[CHIA-1719] Add coin announcement restriction

### PR DESCRIPTION
**Pull request by Quexington**

### CODE REVIEW

File: chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+(mod
+  (
+    Conditions
+  )
+
+  (include condition_codes.clib)
+
+  (defun check_conditions (conditions)
+    (if (= (f (f conditions)) ASSERT_COIN_ANNOUNCEMENT)
```
Comment (matt-o-how) [inline chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp:9]: I understand that this is failing if `(f (f conditions))` is called when `conditions` is empty, but might be worth adding a comment explaining this

File: chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+(mod
```
Comment (matt-o-how) [inline chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp:1]: add a comment to the top of the file explaining what this file does?

File: chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp (hunk 1 above)
Comment (Quexington) [reply chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp:14]: done

File: chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp (hunk 2 above)
Comment (Quexington) [reply chia/wallet/puzzles/custody/restriction_puzzles/wrappers/force_assert_coin_announcement.clsp:5]: done

**Pull request merged**